mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        'amplitude_fit': amplitude_fit,
+        'amplitude_fit_err': amplitude_fit_err,
+        'data_hist_freq': data_hist_freq,
+        'reduced_chi2': reduced_chi2
+        }
+        return dictionary_of_fit_results
+
+    #fitting with composite gaussian lorentzian resolution function and the scatter fractions fixed for some gas species
+    def make_spectrum_composite_gaussian_lorentzian_fixed_survival_probability_partially_fixed_scatter_proportion(self, scatter_proportion, sigma, emitted_peak='shake'):
+        p = scatter_proportion + tuple([1-sum(scatter_proportion)-sum(self.scatter_proportion_for_fixed_gases)]) + tuple(self.scatter_proportion_for_fixed_gases)
+        a = self.recon_eff_param_a
+        b = self.recon_eff_param_b
+        c = self.recon_eff_param_c
+        survival_prob = self.survival_prob
+        scatter_spectra_file_path = os.path.join(self.path_to_scatter_spectra_file, 'scatter_spectra.npy')
+        scatter_spectra = np.load(scatter_spectra_file_path, allow_pickle = True)
+        en_array = self.std_eV_array()
+        current_full_spectrum = np.zeros(len(en_array))
+        if emitted_peak == 'lorentzian':
+            current_working_spectrum = self.std_lorenztian_17keV()
+        elif emitted_peak == 'shake':
+            current_working_spectrum = self.shakeSpectrumClassInstance.shake_spectrum()
+        current_working_spectrum = self.convolve_composite_gaussian_lorentzian(current_working_spectrum, sigma)
+        zeroth_order_peak = current_working_spectrum
+        current_full_spectrum += zeroth_order_peak
+        N = len(self.gases)
+        for M in range(1, self.max_scatters + 1):
+            relative_reconstruction_eff = np.exp(-b*M**c)
+            gas_scatter_combinations = np.array([np.array(i) for i in product(range(M+1), repeat=N) if sum(i)==M])
+            for combination in gas_scatter_combinations:
+                #print(combination)
+                entry_str = ''
+                for component, gas_type in zip(combination, self.gases):
+                    entry_str += gas_type
+                    entry_str += str(component).zfill(2)
+                current_working_spectrum = scatter_spectra.item()[entry_str]
+                current_working_spectrum = self.normalize(signal.convolve(zeroth_order_peak, current_working_spectrum, mode='same'))
+                coefficient = factorial(sum(combination))
+                for component, i in zip(combination, range(N)):
+                    coefficient = coefficient/factorial(component)*p[i]**component
+                for i in range(0, M):
+                    coefficient = coefficient*(1-a*np.exp(-b*i**c))
+                current_full_spectrum += relative_reconstruction_eff*coefficient*current_working_spectrum*survival_prob**M
+        return current_full_spectrum
+
+    def spectrum_func_composite_gaussian_lorentzian_fixed_survival_probability_partially_fixed_scatter_proportion(self, bins_Hz, eff_array, *p0):
```

Comment:
add description of function and p0

Reply:
Added the description of the function and p0

Reply:
@casesyh , I see it says "The amplitude that's scaled so that it's scaled at the level close to the total counts."

Does that mean that the amplitude is about equal to the total number of counts in the scatter peak? Or is the amplitude equal to the height of the zero-th order (un-scattered) peak, in counts?

Reply:
@taliaweiss The fitted amplitude scaled to match the total number of counts in the recorded histogram. In that way we were able to see if the fitted amplitude makes sense. It was a sanity check for testing the line shape model.

Reply:
@casesyh , thank you. Once the fitted amplitude scales to match the total number of counts, does the amplitude then approximately equal the total number of counts in the histogram?

Reply:
@taliaweiss Yes, that's right. The the scaling implemented in the code is intended to make the fitted amplitude reflect the total number of counts in the histogram.